src/SharedUtilities/Domains.jl

```diff
@@ -281,8 +253,65 @@ function coordinates(
     return cc
 end
 
+
+### Example of component specific domain
+"""
+    AbstractVegetationDomain{FT} <: AbstractDomain{FT}
+
+An abstract type for vegetation specific domains.
+"""
+abstract type AbstractVegetationDomain{FT} <: AbstractDomain{FT} end
+
+
+"""
+   RootDomain{FT} <: AbstractVegetationDomain{FT}
+
+Domain for a single bulk plant with roots of varying depths. The user needs
+to specify the depths of the root tips as wel as the heights of the
+compartments to be modeled within the plant. The compartment heights
+are expected to be sorted in ascending order.
+"""
+struct RootDomain{FT} <: AbstractVegetationDomain{FT}
+    "The depth of the root tips, in meters"
+    root_depths::Vector{FT}
+    "The height of the stem, leaf compartments, in meters"
+    compartment_heights::Vector{FT}
+end
+
+function coordinates(domain::RootDomain{FT}) where {FT}
+    return domain.compartment_heights
+end
+
+
+struct SingleColumnLSM{FT} <: AbstractDomain{FT}
+    subsurface::Column{FT}
+    surface::Point{FT}
+end
+
+function SingleColumnLSM(
```

Comment:
I really like this and the way it is used in the `pond_soil_lsm` test. Should we change the name to make it more clear that this has to do with Domains? maybe `SingleColumnDomain` or `SingleColumnLSMDomain`? Otherwise I think it is confusing, since it is not building the full LSM which I think the name implies.

Reply:
good idea! agreed that we should change the name to be more clear.

Reply:
I don't know if there is also a way to make it clearer that this is creating the domains for multiple models, a Column + Point domain, e.g.

The "LSM" addition might be enough. I had to remind myself that this was setting up more than a `Column` domain, though.